examples/energy-management-app/energy-management-common/src/EVSEManufacturerImpl.cpp

```diff
@@ -104,6 +107,224 @@ CHIP_ERROR EVSEManufacturer::Shutdown()
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR FindNextTarget(const uint8_t dayOfWeekMap, uint16_t minutesPastMidnightNow_m, uint16_t & targetTimeMinutesPastMidnight_m,
+                          DataModel::Nullable<Percent> & targetSoC, DataModel::Nullable<int64_t> & addedEnergy_mWh,
+                          bool bAllowTargetsInPast)
+{
+
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    EnergyEvse::Structs::ChargingTargetScheduleStruct::Type entry;
+
+    uint16_t minTimeToTarget_m = 24 * 60; // 24 hours
+    bool bFound                = false;
+
+    EVSEManufacturer * mn = GetEvseManufacturer();
+    VerifyOrReturnError(mn != nullptr, CHIP_ERROR_UNINITIALIZED);
+
+    EnergyEvseDelegate * dg = mn->GetEvseDelegate();
+    VerifyOrReturnError(dg != nullptr, CHIP_ERROR_UNINITIALIZED);
+
+    const DataModel::List<const EnergyEvse::Structs::ChargingTargetScheduleStruct::Type> & chargingTargetSchedules =
+        dg->GetEvseTargetsDelegate()->GetTargets();
+    for (auto & chargingTargetScheduleEntry : chargingTargetSchedules)
+    {
+        if (chargingTargetScheduleEntry.dayOfWeekForSequence.GetField(static_cast<TargetDayOfWeekBitmap>(kAllDays)) & dayOfWeekMap)
+        {
+            // We've found today's schedule - iterate through the targets on this day
+            for (auto & chargingTarget : chargingTargetScheduleEntry.chargingTargets)
+            {
+                if ((chargingTarget.targetTimeMinutesPastMidnight < minutesPastMidnightNow_m) && (bAllowTargetsInPast == false))
+                {
+                    // This target is in the past so move to the next if there is one
+                    continue;
+                }
+
+                if (chargingTarget.targetTimeMinutesPastMidnight < minTimeToTarget_m)
+                {
+                    // This is the earliest target found in the day's targets so far
+                    bFound            = true;
+                    minTimeToTarget_m = chargingTarget.targetTimeMinutesPastMidnight;
+
+                    targetTimeMinutesPastMidnight_m = chargingTarget.targetTimeMinutesPastMidnight;
+
+                    if (chargingTarget.targetSoC.HasValue())
+                    {
+                        targetSoC.SetNonNull(chargingTarget.targetSoC.Value());
+                    }
+                    else
+                    {
+                        targetSoC.SetNull();
+                    }
```

Comment:
```suggestion
                    targetSoC = chargingTarget.targetSoC;
```

Is this the same? It looks like if it has a value, you set it to thyat value, otherwise you set it to null.
Same for below.

Reply:
targetSoc is of type DataModel::Nullable<Percent>
chargingTarget.targetSoC is of type Optional<chip::Percent>
And there is no assignment operator between these two types.